### PR TITLE
#233 implementration + tests

### PR DESCRIPTION
**Pull request by ekostadinov**

### CODE REVIEW

**Comment by iluwatar** [comment]
- the class diagram is non-standard (see [How to contribute](https://github.com/iluwatar/java-design-patterns/wiki/1.-How-to-contribute))
- the context-object module is not included in the super pom.xml

File: context-object/index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+folder: composite
+permalink: /patterns/context-object/
+categories: Structural
+tags: Java
+---
+
+**Intent:** t's purpose is to decouple, as much as possible,
+application data and logic from protocol/presentation-technology
+specific classes such as HttpSession and HttpRequest.
+
```
Comment (iluwatar) [inline context-object/index.md]: > t's purpose is to decouple, as much as possible, application data and logic from protocol/presentation-technology specific classes such as HttpSession and HttpRequest.

Can we improve the phrasing here?

File: context-object/pom.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        <groupId>com.iluwatar</groupId>
+        <artifactId>java-design-patterns</artifactId>
+        <version>1.6.0</version>
+    </parent>
+    <artifactId>context-object</artifactId>
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
```
Comment (iluwatar) [inline context-object/pom.xml]: Remove `version` here. It comes from the super `pom.xml`.

File: context-object/pom.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-all</artifactId>
+            <version>1.3</version>
```
Comment (iluwatar) [inline context-object/pom.xml]: You should introduce the `hamcrest` dependency in the `dependencyManagement` section of the super `pom.xml` and reference it here without the version element. For an example you can check how `junit` dependency is used in the super `pom.xml` and in the submodules.

**Comment by iluwatar** [comment]
I will review it again once you make the corrections add the main class that runs the example. Please make sure to comment the main class extensively so that it is easy to see what the example does based on the description.

**Comment by ekostadinov** [comment]
@iluwatar  can you take a look again?

**Comment by iluwatar** [comment]
- why are you changing abstract-factory/abstract-factory.eml?
- the class diagram is not ok
- you should catch up your branch from origin/master, there are merge conflicts

**Comment by ekostadinov** [comment]
@iluwatar 
 - my Git for Windows don't show merge errors
- abstract-factory/abstract-factory.eml is changed from the Git or IntelliJ (automatically)
- can you help me with the UML ?

**Comment by iluwatar** [comment]
We close pull requests with no activity during the last 30 days.

**Pull request closed without merging**